### ISSUE TITLE
[PROPOSAL] : add `extra_signals` support to `rasterization_2dgs`

### ISSUE THREAD

**Issue by yunusstalha** [nerfstudio-project/gsplat#998]
## Feature request

`gsplat.rendering.rasterization()` supports rendering auxiliary per-Gaussian channels through `extra_signals` (including optional SH evaluation) and returns them separately in `meta["render_extra_signals"]`.

`rasterization_2dgs()` currently has no equivalent interface. Although additional channels can be packed into `colors`, that conflates appearance with auxiliary outputs and makes it harder to preserve a clear RGB/depth output contract, use independent SH settings, or consume the auxiliary render separately.

## Motivation / use case

I would like to render per-Gaussian auxiliary signals alongside the normal 2DGS render, for example learned feature embeddings, semantic logits, or other supervision signals. These signals should use the same visibility ordering and alpha compositing as the 2DGS color render, while remaining available as a distinct output with gradients.

## Proposed API

A possible API would mirror the 3DGS rasterizer:

```python
rasterization_2dgs(
    ...,
    extra_signals: Tensor | None = None,
    extra_signals_sh_degree: int | None = None,
)
```

with output available as:

```python
meta["render_extra_signals"]  # [..., C, H, W, E]
```

Potentially, the expected layouts and semantics could follow `rasterization()`:

* post-activation signals: `[..., (C,) N, E]`
* SH signals: `[N, K, E]`
* compositing consistent with the 2DGS color path
* no behavior or performance regression when `extra_signals is None`

## Implementation direction

My initial thought is to evaluate/normalize the auxiliary signals independently, concatenate them with render channels for the rasterization pass where practical, then split them into `meta["render_extra_signals"]` afterward. I would also add tests for:

* output equivalence when no extra signals are supplied;
* expected signal rendering for batched cameras and multiple signal channels;
* gradients with respect to signals and Gaussian parameters;
* interaction with RGB and depth render modes;
* SH-based auxiliary signals, if that is in scope.

Before I start implementing this, would maintainers prefer API parity with `rasterization()`, or is there a different 2DGS-specific interface/design you would prefer? I would be happy to prepare a PR following the preferred direction.

**Comment by HarperZ9** [comment]
This API shape makes sense to me. The part I would make explicit in the first PR is the receipt/round-trip contract for the auxiliary signal, not only the tensor output.

A small acceptance surface could be:

- `extra_signals=None` produces byte/close numeric equivalence with the current `rasterization_2dgs()` path;
- metadata records signal layout, channel count, SH degree if used, compositing mode, and whether the signal was post-activation or SH-evaluated;
- `meta["render_extra_signals"]` shape is tested for single camera, batched cameras, multiple channels, and RGB/depth render modes;
- RGB/depth outputs remain separate from auxiliary outputs, so packed colors do not become the compatibility path;
- gradients are checked for both signal values and Gaussian parameters where expected;
- one tiny deterministic fixture records render stats such as input gaussian count, signal channel count, output shape, and output hash/tolerance.

The hash/tolerance fixture is useful because auxiliary channels tend to get reused as supervision or measurement inputs later. A reviewer can then tell whether a future change broke visibility/compositing semantics, gradient flow, or only harmless floating-point details.

**Comment by yunusstalha** [comment]
Thanks @HarperZ9. I tried to add the capability, with your suggested tests. Would love to get some feedback.
